src/notifications.js

```diff
@@ -47,27 +54,47 @@ class Notifications extends React.Component {
   }
 
   render() {
-    const {notifications, ...rest} = this.props;
+    const {notifications, store, ...rest} = this.props;
 
     return (
-      <NotifySystem ref='notify' { ...rest } />
+      <NotifySystem ref={this.notifyRef} { ...rest } />
     );
   }
 }
 
 Notifications.propTypes = {
-  notifications: PropTypes.array
+  notifications: PropTypes.array,
+  store: PropTypes.shape({
+    dispatch: PropTypes.func.isRequired
+  }).isRequired
+};
+
+const NotificationsWithContext = props => {
+  const Context = props.context || ReactReduxContext;
+
+  if (Context == null) {
+    throw 'Please upgrade to react-redux v6';
```

Comment:
We are going to release a major version as this + dependency increments could break usage for many consumers.